lua/entities/gmod_wire_adv_microphone.lua

```diff
@@ -0,0 +1,206 @@
+end
+
+function ENT:SpeakerDeactivated(speaker)
+    if self:GetActive() then
+        local live = true
+        do
+            local spk = self._activeSpeakers
+
+            local k1 = next(spk)
+            if k1 == nil then -- No active speakers
+                live = false
+            else
+                local k2 = next(spk)
+                if k2 == nil and k1 == speaker then -- The only active speaker is 'speaker'
+                    live = false
+                end
+            end
+        end
+
+        -- Must be updated before ._activeSpeakers are updated
+        self:SetLive(live)
+    end
+    self._activeSpeakers[speaker] = nil
+end
+
+function ENT:OnRemove()
+    timer.Simple(0, function()
+        if IsValid(self) then return end
+
+        self:SetLive(false)
+    end)
+end
+
+local CVAR_snd_refdb = GetConVar("snd_refdb")
+local CVAR_snd_refdist = GetConVar("snd_refdist")
+
+local function CalculateDistanceGain(dist, sndlevel)
+    -- See SNDLVL_TO_DIST_MULT in engine/audio/private/snd_dma.cpp
+    -- See SND_GetGainFromMult in engine/sound_shared.cpp
+
+    local finalsndlevel = CVAR_snd_refdb:GetFloat() - sndlevel 
+    local distMul = math.pow(10, finalsndlevel / 20) / CVAR_snd_refdist:GetFloat()
+
+    local gain = 1/(distMul * dist)
+
+    return math.min(gain, MAX_DIST_GAIN) -- No infinities
+end
+
+
+
+
+hook.Add("EntityEmitSound", "Wire.AdvMicrophone", function(snd)
+    for _, mic in ipairs(LiveMics) do
+        mic:HandleEngineSound(snd)
+    end
+end)
+
+function ENT:HandleEngineSound(snd)
+    local volume = snd.Volume
+
+    if IsValid(snd.Entity) 
+        and snd.Entity:GetType() == "gmod_wire_adv_speaker" 
+    then 
+        volume = volume * SPEAKER_VOLUME_COEFF
+    end
+
+    local sndlevel = snd.SoundLevel
+    if sndlevel ~= 0 then
+        -- Over-256 values are 'reserved for sounds using goldsrc compatibility attenuation'
+        -- I don't care about correct attenuation for HLSource entities,
+        -- but I don't want the system to break. 
+        if sndlevel >= 256 then sndlevel = sndlevel - 256 end
+        
+        volume = volume * CalculateDistanceGain(
+            self:GetPos():Distance(snd.Pos), sndlevel)
+    end
+    if volume < MIN_VOLUME then return end
+
+    self:ReproduceSound(snd.SoundName, volume, snd.Pitch, snd.DSP)
+end
+
+function ENT:ReproduceSound(snd, vol, pitch, dsp)
+    for _, speaker in ipairs(_activeSpeakers) do
+        speaker:ReproduceSound(snd, vol, pitch, dsp)
+    end
+end
+
+hook.Add("PlayerCanHearPlayersVoice", "Wire.AdvMicrophone", function(listener, talker)
+    local talkerPos = talker:GetPos()
+    local listenerPos = listener:GetPos()
+
+    local speakers = {}
+    -- Note: any given speaker can only be connected to one microphone,
+    -- so this loops can be considered O(nMic), not O(nMic*nSpeaker)
+    for _, mic in ipairs(LiveMics) do
+        if mic:GetPos():DistToSqr(talkerPos) > PLAYER_VOICE_MAXDIST_SQR then continue end
```

Comment:
Can't use continue in Wiremod.

Reply:
![изображение](https://github.com/wiremod/wire/assets/24942438/6dc93d4b-3889-45c5-aef0-68d537b112ad)
Yet it used elsewhere (ignore strings and comments).
Also, without `continue` and `goto`s, there will be four levels of nesting, it will hardly readable.

Reply:
Gotos are fine, but continues are a Garry's addition and not allowed according to the [style guide](https://github.com/wiremod/wire/wiki/Developer-Style-Guide#other-stuff). Those other examples are just outdated code, but new additions like this should conform to the guide.

Reply:
> Yet it used elsewhere (ignore strings and comments).

As denneisk said that's old code. C ops and comments are used everywhere but are being linted to be converted to their lua versions